src/main.js

```diff
@@ -355,11 +355,14 @@ class MiniGraphCard extends LitElement {
     const { show_legend_state = false } = this.config.entities[index];
 
     if (show_legend_state) {
-      if (this.computeUom(index) === '%') {
-        legend += ` (${this.computeState(state)}${this.computeUom(index)})`;
-      } else {
-        legend += ` (${this.computeState(state)} ${this.computeUom(index)})`;
+      legend += ` (${this.computeState(state)}`;
+      if (state !== 'unavailable') {
+        const uom = this.computeUom(index);
+        if (uom !== '%' && uom !== '')
```

Comment:
As we might find more cases where no whitespace is needed, should we generalize to some "included-in-list" logic?

I am also finding the nested if's a bit bulky. Could maybe something like this look a bit tidier?

```js
const uom = this.computeUom(index);
legend += ` (${this.computeState(state)}`;
if (!isNaN(+state))
  legend += ['%', ''].includes(uom) ? `${uom}` : ` ${uom}`
legend += ')';

Reply:
Ok, I'll go with your judgement. 👍